03-rnaseq/differential-expression_rnaseq_01.Rmd

```diff
@@ -0,0 +1,437 @@
+genes_to_keep <- rowSums(counts(ddset)) >= 10
+ddset <- ddset[genes_to_keep, ]
+```
+
+## Run differential expression analysis
+
+We'll use the wrapper function `DESeq()` to do our differential expression analysis.
+In our `DESeq2` object we designated our `asxl_mutation_status` variable as the `model` argument. 
+Because of this, the `DESeq` function will use groups defined by `asxl_mutation_status` to test for differential expression. 
+
+```{r}
+deseq_object <- DESeq(ddset)
+```
+
+Let's extract the results table from the `DESeq` object. 
+
+```{r}
+deseq_results <- results(deseq_object)
+```
+
+Here we will use `lfcShrink()` function to obtain "shrunk" log fold change estimates based on negative binomial distribution. 
+This will add the estimates to your results table. 
+Using `lfcShrink()` can help decrease noise and preserve large differences between groups (it requires that `apeglm` package be installed)
+
+```{r}
+deseq_results <- lfcShrink(deseq_object, # This is the original DESeq2 object with DESeq() already having been ran
+  coef = 2, # This is based on what log fold change coefficient was used in DESeq(), the default is 2.
+  res = deseq_results # This needs to be the DESeq results table
+)
+```
+
+Now let's take a peak at what our results table looks like. 
+
+```{r}
+head(deseq_results)
+```
+
+Note it is not filtered or sorted, so we will use tidyverse to do this before saving our results to a file. 
+Sort and filter the results. 
+
+```{r}
+# this is of class DESeqResults -- we want a data frame
+deseq_df <- deseq_results %>%
+  # make into data.frame
+  as.data.frame() %>%
+  # the gene names are rownames -- let's make this it's own column for easy
+  # display
+  tibble::rownames_to_column("Gene") %>%
+  dplyr::mutate(threshold = padj < 0.05) %>%
+  # let's sort by statistic -- the highest values should be what is up in the
+  # ASXL mutated samples
+  dplyr::arrange(dplyr::desc(log2FoldChange))
+```
+
+Let's print out what the top results are. 
+
+```{r}
+head(deseq_df)
+```
+
+### Check results by plotting one gene
+
+To double check what a differentially expressed gene looks like, we can plot one with `DESeq2::plotCounts()` function.
+
+```{r}
+plotCounts(ddset, gene = "ENSG00000196074", intgroup = "asxl_mutation_status")
+```
+
+The `mutation` group samples have higher expression of this gene than the control group, which helps assure us that the results are showing us what we are looking for. 
+
+## Save results to TSV 
+
+Write the results table to file.
+
+```{r}
+readr::write_tsv(
+  deseq_df,
+  file.path(
+    results_dir,
+    "SRP078441_differential_expression_results.tsv" # Replace with a relevant output file name
+  )
+)
+```
+
+## Create a volcano plot
+
+We'll use the `EnhancedVolcano` package's main function to plot our data. 
+
+```{r}
+volcano_plot <- EnhancedVolcano::EnhancedVolcano(
+  deseq_df,
+  lab = rownames(deseq_df),
+  x = "log2FoldChange",
+  y = "pvalue"
```

Comment:
Should this be `padj`? I remember the default cutoff for the plot being smaller than I might expect for a corrected p-value. (It's also going to depend a lot on what you're trying to accomplish with the plot of course.) But it makes sense to me to be _very clear_ about this value in this setting.